runtime-decompiler/src/main/java/org/jrd/backend/data/ArchivePathManager.java

```diff
@@ -0,0 +1,68 @@
+package org.jrd.backend.data;
+
+import java.util.ArrayList;
+
+public class ArchivePathManager {
+	private static  ArchivePathManager singleton = null;
+
+	public static ArchivePathManager getInstance() {
+		if (singleton == null) {
+			singleton = new ArchivePathManager();
+		}
+		return singleton;
+	}
+
+	private String clazz = "";
+	private boolean found = false;
+	private boolean extracted = false;
+	private ArrayList<String> currentPath = new ArrayList<>();
+
+	public void clear() {
+		clazz = "";
```

Comment:
Once you need method clear, that mens there is something very wrong. Instead of cleared old object, use always new object.